julia-repl.el

```diff
@@ -381,7 +381,9 @@ Queries and appends missing information if necessary.
 Note: when cannot capture the base dir, it is set to NIL to
 prevent further attempts."
   (unless (plist-member (cddr executable-record) :basedir)
-    (let* ((executable-path (cl-second executable-record))
+    (let* ((executable (cl-second executable-record))
+           (executable-path (if (file-name-absolute-p executable) executable
+                              (executable-find executable)))
```

Comment:
This looks nice, but since we are using this code in two places, I would recommend facroring out

```emacs-lisp
(defun julia-repl--executable-path (executable-record)
  "FIXME add docstring"
  (let* ((executable (cl-second executable-record)))
    (if (file-name-absolute-p executable)
        executable
      (executable-find executable))))
```
and just calling it in both places.